syntax = "proto3";

service SCUC{
    rpc GetSCUCResults (SCUCinput) returns (SCUCoutput){

    }
}
message SCUCinput{
    string ppc_method_name = 1;
    bytes ppcfile = 2;
    repeated LOAD load_all = 3;
}
message LOAD{
    int32 bus = 1;
    repeated double load_of_a_day = 2;
}
message SCUCoutput{
    string termination_condition = 1;
    double optimal_value = 2;
    repeated STATUS status = 3;
    repeated POWER power = 4;
    repeated POWER_FLOW power_flow = 5;
}
message STATUS{
    int32 gen = 1;
    repeated int32 status_of_a_day = 2;
}
message POWER{
    int32 gen = 1;
    repeated double power_of_a_day = 2;
}
message POWER_FLOW{
    int32 f_bus = 1;
    int32 t_bus = 2;
    repeated double power_flow_of_a_day =3;
}